runtime/lib/app-runtime.js

```diff
@@ -77,50 +77,80 @@ AppRuntime.prototype.init = function init () {
   if (this.inited) {
     return Promise.resolve()
   }
+  /** 1. init components. */
   this.componentsInvoke('init')
+  /** 2. init device properties, such as volume and cloud stack. */
   this.initiate()
-  /** set turen to not muted */
+  this.resetCloudStack()
+  /** 3. init services */
   this.component.turen.toggleMute(false)
   this.component.turen.toggleWakeUpEngine(true)
+  this.resetServices()
 
+  /** 4. listen on lifetime events */
   this.component.lifetime.on('stack-reset', () => {
     this.resetCloudStack()
   })
   this.component.lifetime.on('preemption', appId => {
     this.appPause(appId)
   })
-  // initializing the whole process...
-  this.resetCloudStack()
-  this.resetServices()
+
+  /** 5. determines if welcome announcements are needed */
   this.shouldWelcome = !this.isStartupFlagExists()
+  var isFirstBoot = property.get('sys.firstboot.init', 'persist') !== '1'
+  property.set('sys.firstboot.init', '1', 'persist')
+
+  var shouldBreakInit = () => {
```

Comment:
`shouldBreakInit` should be a value, so IMO the function name should be `checkBreakInit()` and `var shouldBreak = checkBreakInit()`?

Reply:
Value of `shouldBreak` might vary through out announcements of welcoming. Thus it can not be fetched once and use forever. It has to be fetched again whenever possible.

Reply:
> Thus it can not be fetched once and use forever. It has to be fetched again whenever possible.

Yea, this is what I mean, that's why I did comment at https://github.com/yodaos-project/yodart/pull/450/files#r255860059, which can be evaluated less.